internal/plugin/service/service/models.go

```diff
@@ -0,0 +1,144 @@
+	Components            types.Set      `tfsdk:"components"`
+	DiskSpaceCap          types.String   `tfsdk:"disk_space_cap"`
+	DiskSpaceDefault      types.String   `tfsdk:"disk_space_default"`
+	DiskSpaceStep         types.String   `tfsdk:"disk_space_step"`
+	DiskSpaceUsed         types.String   `tfsdk:"disk_space_used"`
+	Kafka                 types.Set      `tfsdk:"kafka"`
+	MaintenanceWindowDow  types.String   `tfsdk:"maintenance_window_dow"`
+	MaintenanceWindowTime types.String   `tfsdk:"maintenance_window_time"`
+	Plan                  types.String   `tfsdk:"plan"`
+	Project               types.String   `tfsdk:"project"`
+	ProjectVPCId          types.String   `tfsdk:"project_vpc_id"`
+	ServiceHost           types.String   `tfsdk:"service_host"`
+	ServiceIntegrations   types.Set      `tfsdk:"service_integrations"`
+	ServiceName           types.String   `tfsdk:"service_name"`
+	ServicePassword       types.String   `tfsdk:"service_password"`
+	ServicePort           types.Int64    `tfsdk:"service_port"`
+	ServiceType           types.String   `tfsdk:"service_type"`
+	ServiceURI            types.String   `tfsdk:"service_uri"`
+	ServiceUsername       types.String   `tfsdk:"service_username"`
+	State                 types.String   `tfsdk:"state"`
+	StaticIPs             types.Set      `tfsdk:"static_ips"`
+	Tag                   types.Set      `tfsdk:"tag"`
+	TerminationProtection types.Bool     `tfsdk:"termination_protection"`
+
+	// User configs
+	PgUserConfig               types.Set `tfsdk:"pg_user_config"`
+	CassandraUserConfig        types.Set `tfsdk:"cassandra_user_config"`
+	ElasticsearchUserConfig    types.Set `tfsdk:"elasticsearch_user_config"`
+	OpenSearchUserConfig       types.Set `tfsdk:"opensearch_user_config"`
+	GrafanaUserConfig          types.Set `tfsdk:"grafana_user_config"`
+	InfluxdbUserConfig         types.Set `tfsdk:"influxdb_user_config"`
+	RedisUserConfig            types.Set `tfsdk:"redis_user_config"`
+	MysqlUserConfig            types.Set `tfsdk:"mysql_user_config"`
+	KafkaUserConfig            types.Set `tfsdk:"kafka_user_config"`
+	KafkaConnectUserConfig     types.Set `tfsdk:"kafka_connect_user_config"`
+	KafkaMirrormakerUserConfig types.Set `tfsdk:"kafka_mirrormaker_user_config"`
+	M3dbUserConfig             types.Set `tfsdk:"m3db_user_config"`
+	M3aggregatorUserConfig     types.Set `tfsdk:"m3aggregator_user_config"`
+	FlinkUserConfig            types.Set `tfsdk:"flink_user_config"`
+	ClickhouseUserConfig       types.Set `tfsdk:"clickhouse_user_config"`
+}
+
+type serviceIntegration struct {
+	SourceServiceName types.String `tfsdk:"source_service_name"`
+	IntegrationType   types.String `tfsdk:"integration_type"`
+}
+
+var serviceIntegrationAttrs = map[string]attr.Type{
+	"source_service_name": types.StringType,
+	"integration_type":    types.StringType,
+}
+
+func expandServiceIntegration(_ context.Context, _ diag.Diagnostics, o *serviceIntegration) *aiven.NewServiceIntegration {
```

Comment:
consider renaming function to be more descriptive, e.g., `convertToAivenServiceIntegration`